patroni/postgresql/config.py

```diff
@@ -250,6 +250,31 @@ def write_param(self, param, value):
         self.writeline("{0} = '{1}'".format(param, self.escape(value)))
 
 
+class CaseInsensitiveDict(MutableMapping):
+
+    def __init__(self, data):
+        self._store = OrderedDict()
+        self.update(data)
+
+    def __setitem__(self, key, value):
+        self._store[key.lower()] = (key, value)
+
+    def __getitem__(self, key):
+        return self._store[key.lower()][1]
+
+    def __delitem__(self, key):
+        del self._store[key.lower()]
+
+    def __iter__(self):
+        return (casedkey for casedkey, mappedvalue in self._store.values())
```

Comment:
`cased_key` / `mapped_value` (i.e snake_case)

Reply:
This is copy&paste from the `requests` module.